### PR TITLE
feat(auth): Record account security when password fails

### PR DESCRIPTION
**Pull request by vbudhram**
## Because

- We need to add ability to record a security event when a user enters an invalid password

## This pull request

- Adds security event for password failure on login
- Adds security event for removing totp
- Adds security event for adding totp
- Adds security event for success verify totp code/failure

## Issue that this pull request solves

Closes: https://mozilla-hub.atlassian.net/browse/FXA-4924
Closes: https://mozilla-hub.atlassian.net/browse/FXA-4925
Closes: https://mozilla-hub.atlassian.net/browse/FXA-4926
Closes: https://mozilla-hub.atlassian.net/browse/FXA-4927
Closes: https://mozilla-hub.atlassian.net/browse/FXA-4928
Closes https://mozilla-hub.atlassian.net/browse/FXA-4929

## Checklist

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Need to add some tests for this

### CODE REVIEW

File: packages/db-migrations/databases/fxa/patches/patch-136-137.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+SET NAMES utf8mb4 COLLATE utf8mb4_bin;
+
+CALL assertPatchLevel('136');
+
+INSERT INTO securityEventNames(name) VALUES ('account.login.failure');
```
Comment (dschom) [inline packages/db-migrations/databases/fxa/patches/patch-136-137.sql:5]: Should there be a separate PR with all the new event names? Otherwise we might end up with lots of patches.

Comment (vbudhram) [reply packages/db-migrations/databases/fxa/patches/patch-136-137.sql:5]: For sure, this was just a first cut to see if I could get it working. I'll add them in.

File: packages/fxa-auth-server/lib/account-events.ts [hunk 2]
```diff
@@ -68,11 +81,35 @@ export class AccountEventsManager {
...
   }
+
+  /**
+   * Record a security event for the user. This is based on our security events
+   * that are stored in MySQL.
+   *
+   * @param db - auth db
+   * @param message - message
+   */
+  public recordSecurityEvent(db: any, message: SecurityEvent) {
```
Comment (vbudhram) [inline packages/fxa-auth-server/lib/account-events.ts:99]: I kind of went back and forth with whether or not to pass in the db or have it as part of the constructor. Doing the latter was going to have alot of testing churn. Also considering we might do  https://mozilla-hub.atlassian.net/browse/FXA-6268.

File: packages/db-migrations/databases/fxa/patches/patch-137-136.sql [hunk 3]
```diff
@@ -0,0 +1,21 @@
+-- -- Delete new row from table
+-- DELETE FROM securityEventNames WHERE name='account.login.failure';
+-- DELETE FROM securityEventNames WHERE name='account.two_factor_added';
+-- DELETE FROM securityEventNames WHERE name='account.two_factor_requested';
```
Comment (pdehaan) [inline packages/db-migrations/databases/fxa/patches/patch-137-136.sql:4]: Interesting, I think we have 17 delete lines but only 16 insert lines. I think my quick (yet unreliable) eyeballs spotted `'account.two_factor_requested'` as possibly being deleted but not inserted.

File: packages/fxa-auth-server/lib/account-events.ts (hunk 2 above)
Comment (dschom) [reply packages/fxa-auth-server/lib/account-events.ts:99]: Makes sense to me. Random nit,  can we have a type for db? It can even just be an inline type for a db that supports the securityEvent method?

**Review by dschom — APPROVED** [review]
Things look good to me. Thanks for laying the ground work for these tickets! I am going pre-emptively **r+** since the changes look good, and I assume getting the tests to pass isn't a big deal. 

I am pretty sure it's all boils down to a missing Container.reset() in a test suite somewhere.
